src/main/action/index.ts

```diff
@@ -1,34 +1,55 @@
-import DeployNowApi from './api/deployNow';
+import DeployNowClient from './api/deployNow';
 import Types from './input/types';
 import { DisabledProject, ProjectInfo } from './output/types';
+import Retryable from './api/retry';
+import { BranchOverview } from './api/api';
 import { warning } from '@actions/core';
 
 export async function retrieveProjectInfo(configuration: Types): Promise<ProjectInfo | DisabledProject> {
-  const deployNowApi = new DeployNowApi(configuration.serviceHost, configuration.apiKey);
+  const client = new DeployNowClient(configuration.serviceHost, configuration.apiKey);
 
-  const project = await deployNowApi.getProject(configuration.projectId);
-  const isProductionBranch = project.productionBranch.name === configuration.branchName;
-  const branchOverview = await (async () => {
-    if (isProductionBranch) {
-      return project.productionBranch;
-    } else {
-      return await deployNowApi.findBranch(configuration.projectId, configuration.branchName);
-    }
-  })();
+  const branch = await new Retryable<BranchOverview>(
+    async (retry, lastRetry) =>
+      await client.branchApi
+        .getBranches('me', configuration.projectId, { name: configuration.branchName })
+        .then(({ data }) => {
+          if (data.total === 0 || data.values[0].deploymentCount === 0) {
+            if (lastRetry) {
+              return data.values[0];
+            } else {
+              return retry();
+            }
+          }
+          return data.values[0];
+        }),
+    { count: 5 }
+  ).run();
 
-  if (!branchOverview.deploymentEnabled) {
+  if (branch.deploymentCount == 0) {
     warning('The deployment is disabled for this branch');
-    return { 'deployment-enabled': false };
+    return {
+      'deployment-enabled': false,
+    };
   }
 
-  const branch = await deployNowApi.getBranch(configuration.projectId, branchOverview.id);
+  if (branch.deploymentCount > 1) {
+    throw new Error("This action doesn't support multi deployments");
```

Comment:
Add link how to upgrade?

Reply:
If somebody sees this error he has done something strange as it's not possible to create an additional deployment for a project if its workflow isn't migrated.